src/ir_passes/async.ml

```diff
@@ -78,16 +78,9 @@ let new_nary_async_reply ts =
   (* construct the n-ary async value, coercing the continuation, if necessary *)
   let nary_async =
     let coerce u =
-      let v = fresh_var "v" u in
       let k = fresh_var "k" (contT u T.unit) in
-      let r = fresh_var "r" (err_contT T.unit) in
-      [k; r] -->* (
-        varE unary_async -*-
-          (tupE [
-             [v] -->* (varE k -*- varE v);
-             varE r
-          ])
-      )
+      varE (var "@coerce_and_cont" (unary_async --> ([k; fail] -->* (varE unary_async -*- tupE [varE unary_fulfill; varE fail])) |> typ))
```

Comment:
actually this is for 0-ary continuations (impedance matching to unary, with unit)

Can we improve below?

Reply:
Nope, we really get pairs etc., not only unit. Fixing...

Reply:
Is there something wrong with the original code? The replacement actually seems way more convoluted. I would revert, tbh.